stix2/base.py

```diff
@@ -347,6 +360,61 @@ def _check_property(self, prop_name, prop, kwargs):
             for ref in kwargs[prop_name]:
                 self._check_ref(ref, prop, prop_name)
 
+    def _generate_id(self, kwargs):
+        required_prefix = self._type + "--"
+        namespace = uuid.UUID("00abedb4-aa42-466c-9c01-fed23315a9b7")
+
+        try:
+            properties_to_use = self._id_contributing_properties
+            if properties_to_use:
+                streamlined_object = {}
+                if "hashes" in kwargs and "hashes" in properties_to_use:
+                    possible_hash = self._choose_one_hash(kwargs["hashes"])
+                    if possible_hash:
+                        streamlined_object["hashes"] = possible_hash
+                for key in kwargs.keys():
+                    if key in properties_to_use and key != "hashes":
+                        if type(kwargs[key]) is dict:
```

Comment:
Avoid 'type(...) is <type>'.  Use isinstance(...).  If you don't watch to catch _STIXBase by mistake, choose your order so that check comes before the dict check.  E.g.

if isinstance(x, _STIXBase):
...
elif isinstance(x, dict):
...
else:
...

That also flattens out the if statement.